src/routes/session.rs

```diff
@@ -76,48 +77,43 @@ pub fn create(
         String::new()
     };
 
-    if errors.is_empty() {
-        cookies.add_private(
-            Cookie::build(AUTH_COOKIE, user_id)
-                .same_site(SameSite::Lax)
-                .finish(),
-        );
-        let destination = rockets
-            .flash_msg
-            .clone()
-            .and_then(
-                |(name, msg)| {
-                    if name == "callback" {
-                        Some(msg)
-                    } else {
-                        None
-                    }
-                },
-            )
-            .unwrap_or_else(|| "/".to_owned());
-
-        let uri = Uri::parse(&destination)
-            .map(IntoOwned::into_owned)
-            .map_err(|_| {
-                render!(session::login(
-                    &(conn, &rockets.intl.catalog, None, None),
-                    None,
-                    &*form,
-                    errors
-                ))
-            })?;
-
-        Ok(Flash::success(
+    if !errors.is_empty() {
+        return render!(session::login(&rockets.to_context(), None, &*form, errors)).into();
+    }
+
+    cookies.add_private(
+        Cookie::build(AUTH_COOKIE, user_id)
+            .same_site(SameSite::Lax)
+            .finish(),
+    );
+    let destination = rockets
+        .flash_msg
+        .clone()
+        .and_then(
+            |(name, msg)| {
+                if name == "callback" {
+                    Some(msg)
+                } else {
+                    None
+                }
+            },
+        )
+        .unwrap_or_else(|| "/".to_owned());
+
+    if let Ok(uri) = Uri::parse(&destination).map(IntoOwned::into_owned) {
+        Flash::success(
             Redirect::to(uri),
             i18n!(&rockets.intl.catalog, "You are now connected."),
-        ))
+        )
+        .into()
     } else {
-        Err(render!(session::login(
-            &rockets.to_context(),
+        render!(session::login(
```

Comment:
Isn't it a bit strange that when redirecting to invalid uri, it gives an error? It would probably make more sens that the chain on line 100 does the parsing to, and default to `parse("/").unwrap()`. Also it should probably be a [Origin](https://api.rocket.rs/v0.4/rocket/http/uri/struct.Origin.html) to block redirection to other domain
(this is also true for previous code, it's pretty old, I'm pretty sure we did not do code review last September)

Reply:
do you have a suggestion for improvement?

Reply:
I do but I don't know how to explain it with words, so I'll probably do it myself during the week end

Reply:
well, I'll be happy to review it then, and help with words where necessary

i used to be quite good with words, now i'm procrastinating on writing words, by working on a blogging platform.